controllers/cluster_controller_test.go

```diff
@@ -50,57 +48,59 @@ var _ = Describe("Cluster Reconciler", func() {
 		}
 
 		// Create the Cluster object and expect the Reconcile and Deployment to be created
-		Expect(testEnv.Create(ctx, instance)).ToNot(HaveOccurred())
+		g.Expect(testEnv.Create(ctx, instance)).To(Succeed())
 		key := client.ObjectKey{Namespace: instance.Namespace, Name: instance.Name}
 		defer func() {
 			err := testEnv.Delete(ctx, instance)
-			Expect(err).NotTo(HaveOccurred())
+			g.Expect(err).NotTo(HaveOccurred())
 		}()
 
 		// Make sure the Cluster exists.
-		Eventually(func() bool {
+		g.Eventually(func() bool {
 			if err := testEnv.Get(ctx, key, instance); err != nil {
 				return false
 			}
 			return len(instance.Finalizers) > 0
 		}, timeout).Should(BeTrue())
 	})
 
-	It("Should successfully patch a cluster object if the status diff is empty but the spec diff is not", func() {
+	t.Run("Should successfully patch a cluster object if the status diff is empty but the spec diff is not", func(t *testing.T) {
+		g := NewWithT(t)
+
 		// Setup
 		cluster := &clusterv1.Cluster{
 			ObjectMeta: metav1.ObjectMeta{
 				GenerateName: "test2-",
```

Comment:
Orthogonal to this PR some tests here use `Namespace: "default"` whereas others use `Namespace:    v1.NamespaceDefault`, we should unify this and possibly use a test prefixed namespace for each. We might want to use a different PR for it. Thoughts?

Reply:
Let's take it even a step further? We should generate a new namespace on the fly for each package, so tests cannot collide with each other

Reply:
@vincepri Do we need an additional issue to ensure we do this consistently in all our tests?

Reply:
Yes that would be great

Reply:
I created issue #4629 to track that.